src/api/cvc4cppkind.h

```diff
@@ -0,0 +1,2328 @@
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  APPLY_SELECTOR,
+  /**
+   * Datatype selector application.
+   * Parameters: 1
+   *   -[1]: a datatype term (undefined if mis-applied)
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  APPLY_CONSTRUCTOR,
+  /**
+   * Datatype selector application.
+   * Parameters: 1
+   *   -[1]: a datatype term (defined rigidly if mis-applied)
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  APPLY_SELECTOR_TOTAL,
+  /**
+   * Datatype tester application.
+   * Parameters: 2
+   *   -[1]: a tester term
+   *   -[2]: a datatype term
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  APPLY_TESTER,
+#if 0
+  /* Parametric datatype term.  */
+  PARAMETRIC_DATATYPE,
+  /* type ascription, for datatype constructor applications;
+   * first parameter is an ASCRIPTION_TYPE, second is the datatype constructor
+   * application being ascribed */
+  APPLY_TYPE_ASCRIPTION,
+#endif
+  /**
+   * Operator for a tuple update.
+   * Parameters: 1
+   *   -[1]: the index of the tuple to be updated
+   * Create with:
+   *   mkOpTerm(Kind kind, uint32_t param)
+   */
+  TUPLE_UPDATE_OP,
+  /**
+   * Tuple update.
+   * Parameters: 3
+   *   -[1]: a TUPLE_UPDATE_OP (which references an index)
+   *   -[2]: the tuple
+   *   -[3]: the element to store in the tuple at the given index
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  TUPLE_UPDATE,
+  /**
+   * Operator for a record update.
+   * Parameters: 1
+   *   -[1]: the name of the field to be updated
+   * Create with:
+   *   mkOpTerm(Kind kind, const std::string& param)
+   */
+  RECORD_UPDATE_OP,
+  /**
+   * Record update.
+   * Parameters: 3
+   *   -[1]: a RECORD_UPDATE_OP Term (which references a field)
+   *   -[2]: a record term to update
+   *   -[3]: the element to store in the record in the given field
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  RECORD_UPDATE,
+#if 0
+  /* datatypes size */
+  DT_SIZE,
+  /* datatypes height bound */
+  DT_HEIGHT_BOUND,
+  /* datatypes height bound */
+  DT_SIZE_BOUND,
+  /* datatypes sygus bound */
+  DT_SYGUS_BOUND,
+  /* datatypes sygus term order */
+  DT_SYGUS_TERM_ORDER,
+  /* datatypes sygus is constant */
+  DT_SYGUS_IS_CONST,
+#endif
+
+  /* Separation Logic ------------------------------------------------------ */
+
+  /**
+   * Separation logic nil term.
+   * Parameters: 0
+   * Create with:
+   *   mkSepNil(Sort sort)
+   *   mkTerm(Kind kind, Sort sort)
+   */
+  SEP_NIL,
+  /**
+   * Separation logic empty heap.
+   * Parameters: 2
+   *   -[1]: ???
```

Comment:
-[1]: A term whose type is the location type of the heap we are constraining to be empty.